test/depots/depotDonneesParrainages.spec.js

```diff
@@ -36,4 +36,45 @@ describe('Le dépôt de données des parrainages', () => {
     expect(idUtilisateurParrainRecu).to.be('P1');
     expect(filleulAFinaliseCompteRecu).to.be(true);
   });
+
+  it("délègue à l'adaptateur persistance la mise à jour du parrainage", async () => {
+    let idUtilisateurFilleulRecu;
+    let filleulAFinaliseCompteRecu;
+    adaptateurPersistance = {
+      metsAJourParrainage: async (
+        idUtilisateurFilleul,
+        filleulAFinaliseCompte
+      ) => {
+        idUtilisateurFilleulRecu = idUtilisateurFilleul;
+        filleulAFinaliseCompteRecu = filleulAFinaliseCompte;
+      },
+    };
+
+    const parrainage = new Parrainage({
+      idUtilisateurFilleul: 'F1',
+      idUtilisateurParrain: 'P1',
+      filleulAFinaliseCompte: true,
+    });
+
+    await depot().metsAJourParrainage(parrainage);
+
+    expect(idUtilisateurFilleulRecu).to.be('F1');
+    expect(filleulAFinaliseCompteRecu).to.be(true);
+  });
+
+  it("délègue à l'adaptateur persistance la récupération d'un parrainage selon l'id du filleul", async () => {
+    let idUtilisateurFilleulRecu;
+    const parrainage = new Parrainage({});
+    adaptateurPersistance = {
+      parrainagePour: async (idUtilisateurFilleul) => {
+        idUtilisateurFilleulRecu = idUtilisateurFilleul;
+        return parrainage;
+      },
+    };
+
+    const resultat = await depot().parrainagePour('F1');
+
+    expect(idUtilisateurFilleulRecu).to.be('F1');
+    expect(resultat).to.eql(parrainage);
```

Comment:
Techniquement, l'adaptateur Persistance renvoi des `donnees`, et c'est le `depotDonnees` qui instancie la classe `Parrainage`.

Et du coup on peut explicitement écrire l'assert `expect(resultat).to.be.an(Parrainage)`